pragma-node/src/handlers/entries/get_onchain/mod.rs

```diff
@@ -62,18 +75,25 @@ pub async fn get_onchain(
     .map_err(|db_error| db_error.to_entry_error(&pair_id))?;
 
     // TODO(akhercha): ⚠ gives different result than onchain oracle sometime
-    let last_updated_timestamp =
-        get_last_updated_timestamp(&state.onchain_pool, params.network, raw_data.pair_used)
-            .await
-            .map_err(|db_error| db_error.to_entry_error(&pair_id))?;
+    let last_updated_timestamp = get_last_updated_timestamp(
+        &state.onchain_pool,
+        params.network,
+        raw_data[0].pair_used.clone(),
```

Comment:
Don't use `[0]` but `.first()`

Reply:
done

Reply:
is it really usefull as we ensure before that raw_data is not empty ? because first() return an Option while [0] return the data. The only risk was an empty raw_data but it wont reach this code in this case.